### ISSUE TITLE
Balance - taking sectors when not in the sector

### ISSUE THREAD

**Issue by CitrusWire** [Anuken/Mindustry-Suggestions#1405]
Following on from https://github.com/Anuken/Mindustry/issues/3697 - I'd like to suggest that it shouldn't be possible to win a sector if you're out of sector (OOS).

The problem is that the OOS attack calculation is (understandably) simplistic, and can be easily gamed, either accidentally or intentionally. I have twice accidentally gamed it in one short 1/3rd-the-way-through campaign:
1) the above "bug"
2) I just went back into Tar Fields (which after losing horribly to the guardian previously, I had reverted to a save and then forgotten about, having written it off), and I was immediately told I had just captured the sector! My front line is completely intact and all industry is running smoothly; there's no signs of the decimation that I suffered when I played the same save with the same defences manually.

Proposal:
If the player isn't in the sector, the entire sector is simply "frozen". You get nothing from it, and any attacks are put on hold. As well as addressing the above, this also has a gameplay implication: it reinforces that you can only capture sectors with the hands-on approach, rather than just build a few buildings, leave, and auto-win.

The alternatives are:
* create a very-complex OOS calculator - seems like a poor use of dev time.
* run the regular calculator when OOS (which I suspect would hurt performance)
* Current situation

**Comment by Anuken** [comment]
> I had reverted to a save

What do you mean by this? Did you simply copy an older save file into the folder after losing the sector? If so, don't. Sector data is partially stored in other files, and manually messing with them may cause strange problems.

---

In my experience, sectors usually will not last very long unless you provide adequate turret coverage. I have yet to have a capture that I would have considered "unreasonable".

Pausing all waves and production would mean that you would have to be present for every single wave in every single sector, which is tedious for low-difficulty sectors.

**Comment by GreyMario** [comment]
I don't know about you, but I _love_ the fact that I can leave sector 210 alone while it's under attack from Overgrowth and have done so for three attacks now.

Praise be to its Meltdown defense.

I also love the fact that sectors continue to produce even when I'm not in them. It makes shipping in a steady supply of resources in a tough area actually work. It also makes endgame researches possible within a reasonable time frame.

If you have legitimate concerns with "gaming" sector survivability, provide a replicable example with fair documentation.

**Comment by CitrusWire** [comment]
> Did you simply copy an older save file into the folder after losing the sector?

Pretty much. I suspected the part about cross-sector saving so I try and only do it without switching between sectors; timestamps are good indicators as to what's changed. Will endeavour not to do it for any bug reports (so as not to introduce those "strange problems").

> Pausing all waves and production would mean that you would have to be present for every single wave in every single sector, which is tedious for low-difficulty sectors.

(Quick thought: isn't that what the advance-next-wave button is for?)

Well, this brings us to the question as to what the aim of the campaign is? I only got to ~1/3rd the way through before resetting so I don't know. Is it to capture everywhere or just the named sectors?
Follow on question: What's the purpose of the numbered sectors? Something to grind through and get more resources? But you already get tons from the named ones and one or two numbered is more than sufficient to augment them. Plus you're suggesting players should basically minimally-win and skip the numbered ones.

Also, how do players discover they can auto-capture sectors? I didn't know until I did it accidentally, and the game doesn't tell you. It's a non-obvious mechanic.

I'd suggest given the advance-waves button exists and works well, if someone wants to grind through waves it'll only take a few minutes; a fraction of the time it takes to setup any economy.

Alternativelly, maybe freeze and disable auto-capture just for named sectors?



> If you have legitimate concerns with "gaming" sector survivability, provide a replicable example with fair documentation.

There's the one in the linked bug report for Ruinous Shores.

**Comment by Anuken** [comment]
> Well, this brings us to the question as to what the aim of the campaign is? I only got to ~1/3rd the way through before resetting so I don't know. Is it to capture everywhere or just the named sectors?

The named sectors are required to progress, everything else is optional.

> Follow on question: What's the purpose of the numbered sectors? Something to grind through and get more resources? But you already get tons from the named ones and one or two numbered is more than sufficient to augment them.

That wasn't my experience. Later into the tech tree, you need more resources to progress. Having extra production helps. Procedural sectors also act as an extra challenge for those that want one.

> isn't that what the advance-next-wave button is for

You still have to be there and wait for the waves to arrive, which is tedious.

> It's a non-obvious mechanic.

It's not really supposed to be.

**Comment by CitrusWire** [comment]
Well, I just got attacked at Fungal Pass. I don't have any weapons more advanced than salvos/lancers, so the guardian that has the Beam-that-takes-out-your-entire-titanium-shielded-front-line-with-one-swipe carved through said front line like a hot knife through butter.
All I had to do was leave the planet, go to another one of my planets, and sure enough, when I come back a couple minutes later the guardian is gone.
